tests/unit/testMigratedServiceConfiguration.js

```diff
@@ -0,0 +1,553 @@
+/**
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+var test = require('tape');
+var lodash = require('lodash');
+var isNumber = lodash.isNumber;
+var Configuration = require('../fixtures/configuration.js');
+var level = process.env.GCLOUD_ERRORS_LOGLEVEL;
+var logger = require('../../src/logger.js')({
+  logLevel: isNumber(level) ? level : 4
+});
+
+test(
+  'A Configuration uses the function name as the service name on GCF ' + 
+    'if the service name is not given in the given config',
+  function(t) {
+    var oldServiceName = process.env.GAE_SERVICE || '';
+    var oldServiceVersion = process.env.GAE_VERSION || '';
+
+    var oldModuleName = process.env.GAE_MODULE_NAME || '';
+    var oldModuleVersion = process.env.GAE_MODULE_VERSION || '';
+
+    var oldFunctionName = process.env.FUNCTION_NAME || '';
+
+    process.env.GAE_SERVICE = 'someModuleName';
+    process.env.GAE_VERSION = '1.0';
+
+    process.env.GAE_MODULE_NAME = 'InvalidName';
+    process.env.GAE_MODULE_VERSION = 'InvalidVersion';
+
+    process.env.FUNCTION_NAME = 'someFunction';
+
+    var c = new Configuration({}, logger);
+    t.deepEqual(c.getServiceContext().service, 'someFunction');
+    // FUNCTION_NAME is set and the user didn't specify a version, and so 
+    // the version should not be defined
+    t.deepEqual(c.getServiceContext().version, undefined);
+
+    process.env.FUNCTION_NAME = oldFunctionName;
+
+    process.env.GAE_MODULE_VERSION = oldModuleVersion;
+    process.env.GAE_MODULE_NAME = oldModuleName;
+
+    process.env.GAE_VERSION = oldServiceVersion;
+    process.env.GAE_SERVICE = oldServiceName;
+
+    t.end();
+  }
+);
+
+test(
+  'A Configuration uses the function name as the service name on GCF ' +
+    'if the service name is not given in the given config ' +
+    'even if the GAE_SERVICE was not set',
+  function(t) {
+    var oldServiceName = process.env.GAE_SERVICE || '';
+    var oldServiceVersion = process.env.GAE_VERSION || '';
+
+    var oldModuleName = process.env.GAE_MODULE_NAME || '';
+    var oldModuleVersion = process.env.GAE_MODULE_VERSION || '';
+
+    var oldFunctionName = process.env.FUNCTION_NAME || '';
```

Comment:
Extract this boiler plate into a standalone `before` function to avoid repetition, and likewise for `after`.